mlrun/runtimes/kubejob.py

```diff
@@ -134,9 +134,7 @@ def deploy(
                 self.status.state = state
         else:
             self.save(versioned=False)
-            ready = build_runtime(
-                self, with_mlrun, mlrun_version_specifier, watch or is_kfp
-            )
+            ready = build_runtime(self, with_mlrun, mlrun_version_specifier, watch)
```

Comment:
@Hedingber this may lead to the kfp pod exiting before the job was completed in case watch=False